src/serializer/Emitter.js

```diff
@@ -164,7 +163,7 @@ export class Emitter {
   endEmitting(
     dependency: string | Generator | Value,
     oldBody: SerializedBody,
-    valuesToProcess: void | Set<AbstractValue | ConcreteValue>,
```

Comment:
I don't see where this all started, but would `AbstractValue | ObjectValue` be most appropriate? I don't quite see how e.g. a concrete number would ever have to be brought into life by a generator entry, and I see lots of feature interaction potential with e.g. CSE if we allow non-object concrete values to be declared.

Reply:
Maybe we should do that in another PR, the only reason I did it here was to make Flow work without adding loads of invariants.

Reply:
Why would you need loads of invariants? Presumably there's just some specific callsite you introduced that caused everything downstream to want a more generic type. That's the callsite where you need to put the invariant.

Reply:
@gaearon Addressed with Nikolai's PR: https://github.com/facebook/prepack/pull/2210. We can tighten up and not need as many invariants now.